pkg/collector/collector.go

```diff
@@ -733,3 +766,58 @@ func openEventExists(openEvent *tracing.OpenEvent, openEvents []OpenCalls) (bool
 
 	return hasSamePath, hasSameFlags
 }
+
+func (cm *CollectorManager) shouldIncludeOpenEvent(openEvent *tracing.OpenEvent, openEvents []OpenCalls) bool {
+	// Check if we exceeded the maximum number of open events.
+	if len(openEvents) > MaxOpenEvents {
+		return false
+	}
+
+	// Check if we should ignore this path.
+	if os.Getenv("OPEN_IGNORE_PREFIXES") != "" {
+		for _, prefix := range strings.Split(os.Getenv("OPEN_IGNORE_PREFIXES"), ",") {
+			if strings.HasPrefix(openEvent.PathName, prefix) {
+				log.Printf("ignoring open event %s because it matches prefix %s\n", openEvent.PathName, prefix)
+				return false
+			}
+		}
+	}
+
+	// Check if event is already in the list.
+	hasSamePath, hasSameFlags := openEventExists(openEvent, openEvents)
+	if hasSamePath && hasSameFlags {
+		return false
+	}
+
+	// Check if we should ignore mounts.
+	if os.Getenv("OPEN_IGNORE_MOUNTS") == "true" {
+		cm.podMountCacheMutex.Lock()
+		mounts := cm.podMountCache[fmt.Sprintf("%s-%s", openEvent.PodName, openEvent.Namespace)]
+		cm.podMountCacheMutex.Unlock()
```

Comment:
maybe move this to the caller function to prevent too many lock

Reply:
Done